src/kilo.c

```diff
@@ -939,13 +939,45 @@ void editorMoveCursorToRowEnd() {
     }
     int rowlen = row ? row->size : 0;
     int size = rowlen - E.cx-E.coloff;
-    
-    while (size != 0 ) {
+
+    while (size != 0) {
         editorMoveCursor(RIGHT);
         size = size -1;
     }
 }
 
+int editorMoveCursorToFirst(char c) {
+  int filerow = E.rowoff+E.cy;
+  erow *row = (filerow >= E.numrows) ? NULL : &E.row[filerow];
+
+  if (row == NULL) {
+    return 0;
+  }
+
+  int rowlen = row ? row->size : 0;
+  int size = rowlen;
+
+  int i = E.cx-E.coloff;
+  int initialOffSet = E.cx-E.coloff;
+  int found = 0;
+
+  while (row->chars[i] != c && i != size) {
+    if (row->chars[i+1] == c) {
+      found = 1;
+    }
+    i++;
+  }
+
+  if (found == 1) {
+    while (i - initialOffSet != 0) {
+      i = i -1;
```

Comment:
Not currently, I wouldn't be opposed to a wrapper around editorMoveCursor that took a length argument though.